service/history/workflow/task_generator.go

```diff
@@ -419,6 +417,7 @@ func (r *TaskGeneratorImpl) GenerateScheduleWorkflowTaskTasks(
 			EventID:             workflowTask.ScheduledEventID,
 			ScheduleAttempt:     workflowTask.Attempt,
 			Version:             workflowTask.Version,
+			Stamp:               r.mutableState.GetExecutionInfo().GetWorkflowTaskStamp(), // must use current stamp!
```

Comment:
oh maybe provide some more context here. Guess I am still confused on why workflow.Stamp would be wrong here?

Reply:
Alright; I took another look and when I wrote this I didn't realize that `GetWorkflowTaskByID` is actually just returning a struct with the current workflow task's info. I thought it returned the _currently_ scheduled workflow task (I didn't know we don't store that). So I've removed the special case here now!